test_agent/java/src/main/java/org/eclipse/uprotocol/TestAgent.java

```diff
@@ -113,19 +125,32 @@ private static void writeDataToTMSocket(JSONObject responseDict, String action)
 
         }
     }
-
+    
     private static void sendToTestManager(Object json, String action) {
+    	// Object json should be str or dict 
         // Create a new dictionary
         JSONObject responseDict = new JSONObject();
         responseDict.put("data", json);
         writeDataToTMSocket(responseDict, action);
 
     }
 
+    private static void sendToTestManager(Object json, String action, String received_test_id) {
```

Comment:
Do we actually need these 4 methods with the same name "sendToTestManager" ?? Can we handle it with just one or two?

Reply:
We believe that using 4 methods is still the best option. Any option that uses one or two would require the caller of the method to explicitly pass null. Instead, we simplified the 4 methods, which hopefully works for you.